### PR TITLE
Collapsible separators - Fix & Improvements

### PR DESCRIPTION
**Pull request by Holt59**
## Improvements

- Re-organize settings and add a new "User Interface" tab with some new options

image

- Display contents from collapsed mods in separators
  - Conflict icons (already in the previous PR)
  - Mod contents + Tooltip
  - Mod flags
  - Version: only the icons for "Update available" and "Warning"
- Display conflicts from mods in collapsed separators when the separator is selected
- Alt+Click to select a separator with its children, collapsed or not.
- Allow double-click to accept the current choice in `ListDialog`.
- Minor visual improvements
  - Icons should now be vertically-centered on the row (contents, conflicts, etc.). This is hard to see on the default theme but more perceptible on theme such as VS Dark where the rows are taller.
  - Fix the "ideal text color" by merging the background with the color before computing the foreground color. Qt actually fills the background of the tree view before the background of each row, and since the conflicts background are semi-transparent, there were huge difference between light and dark theme which were not reflected in the foreground color. This aims to fix it.

## Fixes

- Partial fix for Shift+Click improper selection after editing an item, only works when the item has not been edited.
  - Fixing it more globally is more complex because we need to remember the selection, which is not easy since the name change can trigger a refresh... I'm not sure if it's worth it since I think the error mostly occurs when users trigger the edit by mistake while selecting mod, not when they are actually renaming mods.
- Add the "Create empty mod" and "Create separator" to the global context menu when clicking in a blank area in the mod list or using the toolbar button.
- Fix drop below collapsed separators that would drop at the beginning of the separator instead of at the end.

## Refactoring

- Some minor stuffs not worth detailing here.

### CODE REVIEW

**Review by Al12rs — APPROVED** [review]

**Comment by isanae** [comment]
This will require updating https://github.com/ModOrganizer2/modorganizer/wiki/Settings-window.

File: src/modlist.cpp [hunk 1]
```diff
@@ -585,15 +555,16 @@ bool ModList::renameMod(int index, const QString &newName)
...
       // the modRenamed() call will refresh the indexes in the current profile
       // and update the modlists in all profiles
       emit modRenamed(oldName, nameFixed);
-  }
+    }
 
-  // invalidate the currently displayed state of this list
-  notifyChange(-1);
+    // invalidate the currently displayed state of this list
+    notifyChange(-1);
```
Comment (isanae) [inline src/modlist.cpp:566]: The braces above have changed, so this line is inside the `if` now. Was that intended? Was it the selection fix?

File: src/modlistview.cpp [hunk 2]
```diff
@@ -1232,6 +1355,73 @@ void ModListView::timerEvent(QTimerEvent* event)
   }
 }
 
+void ModListView::mousePressEvent(QMouseEvent* event)
```
Comment (isanae) [inline src/modlistview.cpp:1358]: Can you add a short comment as to what this and `mouseReleaseEvent()` are actually doing? Something like "alt+click selects all mods inside a separator if the option is enabled".

File: src/modlist.cpp (hunk 1 above)
Comment (Holt59) [reply src/modlist.cpp:566]: Yes it's intended and that's the partial Shift+Click selection fix when editing.

**Review by isanae — APPROVED** [review]

**Pull request merged**